File: 20-actors/shionome/methods/test_registry.py
"""test_registry.py — 潮目 (shionome) registry access + sourcing honesty. ADR-2606072200."""
from __future__ import annotations

import registry
from _t import expect_raises, run


def test_source_ids_nonempty():
    assert len(registry.source_ids()) >= 10


def test_get_source_known():
    s = registry.get_source("us-fred")
    assert s["jurisdiction"] == "us"


def test_get_source_unknown_raises():
    expect_raises(lambda: registry.get_source("nope"), contains="no such source")


def test_sourcing_unverified_is_representative():
    assert registry.sourcing_for("us-fred") == ":representative"


def test_sourcing_unknown_is_representative():
    assert registry.sourcing_for("nope") == ":representative"


def test_assert_source_allowed_blocks_terminal():
    expect_raises(lambda: registry.assert_source_allowed("via refinitiv eikon"), contains="Rider")


def test_assert_source_allowed_passes_public():
    registry.assert_source_allowed("https://fred.stlouisfed.org/")


if __name__ == "__main__":
    run("registry", [(n, f) for n, f in sorted(globals().items())
                     if n.startswith("test_") and callable(f)])

File: 20-actors/shionome/methods/test_registry.clj
#!/usr/bin/env bb
;; Clojure port of test_registry.py — 潮目 (shionome) registry access + sourcing honesty.
;; ADR-2606072200.
(ns shionome.methods.test-registry
  (:require [clojure.test :refer [deftest is run-tests testing]]
            [shionome.methods.registry :as registry]))

;; ── tests (7, matching test_registry.py exactly) ────────────────────────────────

(deftest test-source-ids-nonempty
  (testing "source-ids returns at least 10 entries"
    (is (>= (count (registry/source-ids)) 10))))

(deftest test-get-source-known
  (testing "get-source for us-fred returns jurisdiction=us"
    (let [s (registry/get-source "us-fred")]
      (is (= (get s "jurisdiction") "us")))))

(deftest test-get-source-unknown-raises
  (testing "get-source with unknown id throws ex-info containing 'no such source'"
    (is (thrown-with-msg? clojure.lang.ExceptionInfo #"no such source"
                          (registry/get-source "nope")))))

(deftest test-sourcing-unverified-is-representative
  (testing "sourcing-for an unverified-seed source returns :representative"
    (is (= (registry/sourcing-for "us-fred") ":representative"))))

(deftest test-sourcing-unknown-is-representative
  (testing "sourcing-for an unknown source id returns :representative conservatively"
    (is (= (registry/sourcing-for "nope") ":representative"))))

(deftest test-assert-source-allowed-blocks-terminal
  (testing "assert-source-allowed raises on a prohibited commercial terminal (Rider §2(e))"
    (is (thrown-with-msg? clojure.lang.ExceptionInfo #"Rider"
                          (registry/assert-source-allowed "via refinitiv eikon")))))

(deftest test-assert-source-allowed-passes-public
  (testing "assert-source-allowed passes clean public-source text without throwing"
    (is (nil? (registry/assert-source-allowed "https://fred.stlouisfed.org/")))))

;; ── runner ───────────────────────────────────────────────────────────────────────

(when (= *file* (System/getProperty "babashka.file"))
  (let [{:keys [fail error]} (clojure.test/run-tests 'shionome.methods.test-registry)]
    (System/exit (if (zero? (+ fail error)) 0 1))))
